Shared/Samples/Identify raster cell/IdentifyRasterCellView.swift

```diff
@@ -0,0 +1,200 @@
+        ///   - proxy: The proxy used to identify the screen point on the raster layer.
+        /// - Returns: The first raster cell found in the identify result.
+        private func rasterCell(at screenPoint: CGPoint, using proxy: MapViewProxy) async -> RasterCell? {
+            do {
+                // Identify the screen point on the raster layer using the map view proxy.
+                let identifyResult = try await proxy.identify(on: rasterLayer, screenPoint: screenPoint, tolerance: 1)
+                
+                // Get the first raster cell from the identify result.
+                let rasterCell = identifyResult.geoElements.first(where: { $0 is RasterCell })
+                return rasterCell as? RasterCell
+            } catch {
+                self.error = error
+                return nil
+            }
+        }
+        
+        /// Updates the location of the callout placement to a given screen point.
+        /// - Parameters:
+        ///   - screenPoint: The screen point at which to place the callout.
+        ///   - proxy: The proxy used to convert the screen point to a map point.
+        private func updateCalloutPlacement(to screenPoint: CGPoint, using proxy: MapViewProxy) {
+            // Create an offset to offset the callout if needed, e.g. the magnifier is showing.
+            let offset = calloutShouldOffset ? CGPoint(x: 0, y: -70) : .zero
+            
+            // Get the map location of the screen point from the map view proxy.
+            if let location = proxy.location(fromScreenPoint: screenPoint) {
+                calloutPlacement = .location(location, offset: offset)
+            }
+        }
+        
+        /// Updates the text shown in the callout using the attributes and coordinates of a given raster cell.
+        /// - Parameter cell: The raster cell to create the text from.
+        private func updateCalloutText(using cell: RasterCell) {
+            // Create the attributes text using the attributes of the raster cell.
+            let attributes = cell.attributes
+                .map { "\($0.key): \($0.value)" }
+                .sorted(by: >)
+                .joined(separator: "\n")
+            
+            // Create the coordinate texts using the extent of the cell's geometry.
+            guard let extent = cell.geometry?.extent else {
+                calloutText = attributes
+                return
+            }
+            let roundedStyle = FloatingPointFormatStyle<Double>.number.rounded(rule: .awayFromZero, increment: 0.001)
+            let xCoordinate = "X: \(extent.xMin.formatted(roundedStyle))"
+            let yCoordinate = "Y: \(extent.yMin.formatted(roundedStyle))"
+            
+            // Update the callout text.
+            calloutText = "\(attributes)\n\n\(xCoordinate)\n\(yCoordinate)"
+        }
+    }
+    
+    // A generic struct made up of two equatable types.
+    struct EquatablePair<T: Equatable, U: Equatable>: Equatable {
```

Comment:
I was contemplating if we really need this but I guess it is fine because the other solution would be to assign two task view modifiers to the view. One for if `calloutShouldOffset` changes and the other one for `tapScreenPoint` changes. But I'm not sure that is much better.

Reply:
I didn't go with the two task modifiers so that the callout isn't created twice at the start of the long press. Besides that being less efficient/more redundant, doing so isn't async safe and can occasionally cause it to crash when creating two at once. Optimally, I wanted would want it to just watch the `tapScreenPoint`, but I couldn't figure out a way to force that variable to update in the gesture `onEnded` since the point doesn't actually change then. I guess I could make another state variable just for removing the offset and do two `task` modifiers with that if you think that is a good idea.

Reply:
I think what you have will be fine, thanks.